pkg/apis/config/default.go

```diff
@@ -39,6 +41,11 @@ type Defaults struct {
 	DefaultServiceAccount string
 	DefaultRunAsUser      int64
```

Comment:
we dont need these default if we want to keep empty value in cm

Reply:
As mentioned here https://github.com/tektoncd/triggers/pull/1747#discussion_r1667315197 imo if user don't set empty then we can set them with default values 

It shouldn't be mandatory that user has to set values through CM its completely optional (which is the current behavior )

 Only thing is now we are providing way to configure different values by providing those keys in CM or remove default values for those keys if they don't want by setting `""`